File: android/src/main/location-engine-default/org/maplibre/reactnative/location/engine/LocationEngineProvider.java
package org.maplibre.reactnative.location.engine;

import android.content.Context;

import org.maplibre.android.location.engine.LocationEngine;

public class LocationEngineProvider implements LocationEngineProvidable {
    @Override
    public LocationEngine getLocationEngine(Context context) {
        return new DefaultLocationEngineProvider().getLocationEngine(context);
    }
}

File: package/android/src/main/location-engine-google/org/maplibre/reactnative/location/engine/LocationEngineProvider.kt
package org.maplibre.reactnative.location.engine

import android.content.Context
import org.maplibre.android.location.engine.LocationEngine

class LocationEngineProvider : LocationEngineProvidable {
    override fun getLocationEngine(context: Context): LocationEngine = GoogleLocationEngineProvider().getLocationEngine(context)
}
